### PR TITLE
[Impeller] Remove redundant GN flags.

### PR DESCRIPTION
**Pull request by chinmaygarde**
These were added to conditionally compile stuff when Impeller only had Metal on
Darwin implementation. These are no longer used.

### CODE REVIEW

File: impeller/BUILD.gn [hunk 1]
```diff
@@ -9,10 +9,6 @@ config("impeller_public_config") {
 
   defines = []
 
-  if (impeller_supports_platform) {
-    defines += [ "IMPELLER_SUPPORTS_PLATFORM=1" ]
```
Comment (zanderso) [inline impeller/BUILD.gn:13]: Do we need to remove references to this from the code?

Comment (chinmaygarde) [reply impeller/BUILD.gn:13]: Yeah. I built the unit-tests to verify. Going over the code to change these references.

Comment (chinmaygarde) [reply impeller/BUILD.gn:13]: Done.

**Review by bdero — APPROVED** [review]

File: shell/platform/android/platform_view_android.cc [hunk 2]
```diff
@@ -10,14 +10,10 @@
 #include "flutter/fml/synchronization/waitable_event.h"
 #include "flutter/shell/common/shell_io_manager.h"
 #include "flutter/shell/gpu/gpu_surface_gl_delegate.h"
-#if IMPELLER_SUPPORTS_PLATFORM
```
Comment (dnfield) [inline shell/platform/android/platform_view_android.cc:13]: This was done as a hack to make sure that google3 would build.

Can we make sure build_engine=1 still works with this patch applied in google3?

Comment (chinmaygarde) [reply shell/platform/android/platform_view_android.cc:13]: I expect the rules already ported for the Metal backend for ImpellerC to also work with the OpenGL backend (and others).

Comment (zanderso) [reply shell/platform/android/platform_view_android.cc:13]: The g3 build hardcodes `IMPELLER_SUPPORTS_PLATFORM` when depending on impeller. An internal follow-up CL would remove the define from the impeller target.

**Review by zanderso — APPROVED** [review]

**Pull request merged**